### ISSUE TITLE
timer.h issue 

### ISSUE THREAD

**Issue by puskartrital** [Siegfried5/Keylogger#4]
in timer.h this_thread is not declared im not saying compiler said it when i try to run main.cpp
Thread not declared
thread is not a member of std
thread in namespace std does not name a type
![error](https://user-images.githubusercontent.com/31067508/38596502-a02eacaa-3d71-11e8-8201-dbb809480793.png)

**Comment by Siegfried5** [comment]
do you have c++11 standards enabled in code blocks?
To check go to `Setting -> Compiler`
and you should see a box that says 
`[ ]Have g++ follow the C++11 ISO C++  language standard [-std=c++11]`

if it unmarked than you have to toggle it on. Than try compiling main.cpp again and see if it works

**Comment by puskartrital** [comment]
Yes it worked thank you, but in timer.h error again in std:: thread Thread this line i don't know why?
recompile your timer.h if it works in my system doesn't work. Error detected.
.

**Comment by Siegfried5** [comment]
is it the same error?

**Comment by puskartrital** [comment]
no it's not code is wrong or either something missing.

**Comment by Siegfried5** [comment]
I only get errors when i don't have the c++ 11 standards box marked. Other wise I am able to compile with no problem

**Comment by Siegfried5** [comment]
can you send me the errors

**Comment by puskartrital** [comment]
![screenshot at 2018-05-15 20-41-37](https://user-images.githubusercontent.com/31067508/40065040-83e9e4cc-5880-11e8-8266-c35be96e0120.png)
See errors i get in timer.h

**Comment by Siegfried5** [comment]
can you send a pic of your compiler setting. please

**Comment by sunnepazzy123** [comment]
Please i am new in this programming and here in this site..i saw over 5 codes for the email keylogger and i downloaded power.docx and sendmail.h and i read d script..is that all i need or do i still need to add the source code in my SendMail.h code..?

**Comment by Siegfried5** [comment]
Sunnepazy123. Unless you want to change how the sendmail function work the only thing you need to do is add what it says in the readme

**Comment by sunnepazzy123** [comment]
So all i need to use in the source is the sendmail_h source code by adding
the powershell.docx code if the sendmail_h code have issue..

**Comment by sunnepazzy123** [comment]
I really appreciate your quick response because i wasn't expecting it this
soon.thank you Siegfried5

**Comment by Siegfried5** [comment]
Yes. Read the readme. It's will tell you what to add to make the key logger work

**Comment by sunnepazzy123** [comment]
Okay..Thank you so much..i will keep u posted..

**Comment by sunnepazzy123** [comment]
I don't know what u mean here..and how to fix it..pls kindly tell me

added a new flag with the following Setting -> Compiler
name: windowsAPI
Compiler flags: -mwindows
linker flags: -mwindows

**Comment by Siegfried5** [comment]
are you using codeblocks?

**Comment by sunnepazzy123** [comment]
Yes i am

On May 29, 2018 4:53 AM, "Siegfried5" <notifications@github.com> wrote:

are you using codeblocks?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/Siegfried5/Keylogger/issues/4#issuecomment-392645586>,
or mute the thread

**Comment by Siegfried5** [comment]
go to codeblocks setting, than in the setting go to compiler. Their is were you need to create and the compiler flag. Also make sure you turn on the C++ 11 standards( its located in the same place)

**Comment by sunnepazzy123** [comment]
I have turn on the c++ 11 standard..i will send you a snapshot of what i
did..

On May 29, 2018 7:07 AM, "Siegfried5" <notifications@github.com> wrote:

go to codeblocks setting, than in the setting go to compiler. Their is were
you need to create and the compiler flag. Also make sure you turn on the
C++ 11 standards( its located in the same place)

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/Siegfried5/Keylogger/issues/4#issuecomment-392662527>,
or mute the thread

**Comment by Siegfried5** [comment]
I am not getting a screenshot

**Comment by sunnepazzy123** [comment]
The code in codeblock is all green but in the build message it says No SUCH
FILE OR DIRECTORY..

**Comment by Siegfried5** [comment]
Did you create a new C++ Console App. Also did you just drag and drop the files into code block. Can you send Me a screen shot of codeblocks. It will be really helpful.

**Comment by sunnepazzy123** [comment]
I have done as you said in the compiler settings..kindly check my screenshot

**Comment by sunnepazzy123** [comment]
I downloaded the SENDme_h file and the powerscript and i did exactly what d
Readme said..it show all green in my codeblock but refuse to build it
saying -mwindows no such file or directory in the code block build messages
log...i didnt download the rest of those other source code..i hope that is
not the problem..

**Comment by Siegfried5** [comment]
Again I do not see a screen shot on here.

**Comment by ghost** [comment]
and AV detect this ?

**Comment by Siegfried5** [comment]
some av do detect this.

**Comment by pbennyjoseph** [comment]
> do you have c++11 standards enabled in code blocks?
> To check go to `Setting -> Compiler`
> and you should see a box that says
> `[ ]Have g++ follow the C++11 ISO C++ language standard [-std=c++11]`
> 
> if it unmarked than you have to toggle it on. Than try compiling main.cpp again and see if it works


And also make sure CodeBlocks is version 16.01. I completely uninstalled CodeBlocks 17.12 and Installed 
CodeBlocks 16.01 as in the Udemy course. and it Worked with those flags. Newer version of CodeBlocks not compiling even with compiler flags.
